activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -398,18 +398,15 @@ def indexes(table_name, name = nil) #:nodoc:
         indexes
       end
 
-      # Returns an array of +Column+ objects for the table specified by +table_name+.
-      def columns(table_name) # :nodoc:
-        table_name = table_name.to_s
-        column_definitions(table_name).map do |field|
-          type_metadata = fetch_type_metadata(field[:Type], field[:Extra])
-          if type_metadata.type == :datetime && field[:Default] == "CURRENT_TIMESTAMP"
-            default, default_function = nil, field[:Default]
-          else
-            default, default_function = field[:Default], nil
-          end
-          new_column(field[:Field], default, type_metadata, field[:Null] == "YES", table_name, default_function, field[:Collation], comment: field[:Comment].presence)
+      # Creates Column object from the result of SHOW FULL FIELDS
+      def new_column_from_field(table_name, field)
```

Comment:
It does need to be public. We are using at shopify because we have a custom code to dump the schema but the plan is not need this custom code anymore and move it inside Rails. When we do that we will not be using that method outside the framework anymore